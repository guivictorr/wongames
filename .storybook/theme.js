import { create } from '@storybook/theming'
import theme from '../src/styles/theme'

export default create({
  base: 'dark',

  colorPrimary: theme.colors.primary,
  colorSecondary: theme.colors.secondary,

  // UI
  appBg: theme.colors.mainBg,
  appContentBg: theme.colors.mainBg,
  appBorderColor: theme.colors.lightGray,
  appBorderRadius: 4,

  // Typography
  fontBase: '"Open Sans", sans-serif',
  fontCode: 'monospace',

  // Text colors
  textColor: theme.colors.white,
  textInverseColor: theme.colors.primary,

  // Toolbar default and active colors
  barTextColor: theme.colors.white,
  barSelectedColor: theme.colors.primary,
  barBg: theme.colors.mainBg,

  // Form colors
  inputBg: theme.colors.lightGray,
  inputBorder: theme.colors.darkGray,
  inputTextColor: theme.colors.black,

  brandTitle: 'Won games storybook',
  brandUrl: 'https://example.com',
  brandImage: '/img/logo.svg'
})
