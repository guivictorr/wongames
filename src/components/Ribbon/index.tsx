import * as S from './styles'

export type RibbonColors = 'primary' | 'secondary'
export type RibbonSizes = 'small' | 'normal'

export type RibbonProps = {
  color?: RibbonColors
  size?: RibbonSizes
  children: React.ReactNode
}

const Ribbon = ({ children, ...props }: RibbonProps) => (
  <S.Wrapper {...props}>{children}</S.Wrapper>
)

export default Ribbon
