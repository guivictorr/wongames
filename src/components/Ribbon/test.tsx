import { screen } from '@testing-library/react'
import theme from 'styles/theme'
import { renderWithTheme } from 'utils/tests/helpers'

import Ribbon from '.'

describe('<Ribbon />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Ribbon>Best Seller</Ribbon>)

    expect(screen.getByText(/best seller/i)).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render with primary color by default', () => {
    renderWithTheme(<Ribbon>Best Seller</Ribbon>)

    expect(screen.getByText(/best seller/i)).toHaveStyle({
      backgroundColor: theme.colors.primary
    })
  })

  it('should render with secondary color when passed', () => {
    renderWithTheme(<Ribbon color="secondary">Best Seller</Ribbon>)

    expect(screen.getByText(/best seller/i)).toHaveStyle({
      backgroundColor: theme.colors.secondary
    })
  })

  it('should render with normal size as default', () => {
    renderWithTheme(<Ribbon>Best Seller</Ribbon>)

    expect(screen.getByText(/best seller/i)).toHaveStyle({
      height: '3.6rem',
      fontSize: theme.font.sizes.small
    })
  })

  it('should render with small size when passed', () => {
    renderWithTheme(<Ribbon size="small">Best Seller</Ribbon>)

    expect(screen.getByText(/best seller/i)).toHaveStyle({
      height: '2.6rem',
      fontSize: theme.font.sizes.xsmall
    })
  })
})
