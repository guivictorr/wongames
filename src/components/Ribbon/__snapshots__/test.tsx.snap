// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ribbon /> should render correctly 1`] = `
.c0 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #FAFAFA;
  top: 1.6rem;
  font-weight: 600;
  background-color: #F231A5;
  font-size: 1.4rem;
  padding: 0 2.4rem;
  height: 3.6rem;
  right: -2rem;
}

.c0::before {
  content: '';
  position: absolute;
  right: 0;
  border-style: solid;
  border-left-width: 0rem;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-bottom-width: 1rem;
}

.c0::before {
  border-left-color: #b20b6f;
  border-top-color: #b20b6f;
}

.c0::before {
  top: 3.6rem;
  border-top-width: 1rem;
  border-right-width: 2rem;
}

<main
  class="c0"
>
  Best Seller
</main>
`
